### PR TITLE
#202 fix(dropdown): merge valueItemContainerProps more intelligently

### LINKED ISSUE #202: Dropdown | valueContainer props aren't applied to the Button correctly
While we could pass the valueContainerProps to the containerProps of Button - this would overwrite the isOpen data. We need to merge the props.
image

### PR DESCRIPTION
**Pull request by aVileBroker**
fix #202

### CODE REVIEW

File: packages/foundry-react-ui/src/components/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -423,16 +423,19 @@ const Dropdown = ({
...
         onClick={(e: React.MouseEvent) => e.preventDefault()}
         onMouseDown={clickHandler}
         variant={variant}
         containerRef={valueContainerRef}
         {...valueContainerProps}
+        containerProps={{
+          isOpen,
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-expect-error
+          ...(valueContainerProps ? valueContainerProps.containerProps : {}),
```
Comment (aVileBroker) [inline packages/foundry-react-ui/src/components/Dropdown/Dropdown.tsx:437]: It is worried about valueContainerProps being undefined, so I covered for that possibility and threw an expect error on it

**Review by donaldjbrady — APPROVED** [review]

File: packages/foundry-react-ui/src/components/Dropdown/Dropdown.tsx (hunk 1 above)
Comment (joien1) [reply packages/foundry-react-ui/src/components/Dropdown/Dropdown.tsx:437]: I am unsure what the error here would be. If valueContainerProps is undefined, it should choose the second option in the ternary, right?

Comment (aVileBroker) [reply packages/foundry-react-ui/src/components/Dropdown/Dropdown.tsx:437]: Yep - typescript isn't that smart unfortunately. Of course the "right" option would probably be to improve the typing here - but we're using a generic subcomponent type

**Pull request merged**